--- libaio-0.3.111_old/src/Makefile	2019-01-26 10:54:53.200000000 +0800
+++ libaio-0.3.111/src/Makefile	2019-01-26 11:20:44.106000000 +0800
@@ -3,7 +3,7 @@
 libdir=$(prefix)/lib
 
 CFLAGS ?= -g -fomit-frame-pointer -O2
-CFLAGS += -nostdlib -nostartfiles -Wall -I. -fPIC
+CFLAGS += $(RPM_OPT_FLAGS) -nostdlib -nostartfiles -Wall -I. -fPIC
 SO_CFLAGS=-shared $(CFLAGS)
 L_CFLAGS=$(CFLAGS)
 LINK_FLAGS=
